bpx/schema.py

```diff
@@ -319,23 +294,20 @@ class ElectrodeSingleSPM(ContactBase, Particle):
     Particle type models.
     """
 
-    pass
-
 
 class ElectrodeBlendedSPM(ContactBase):
     """
     Class for electrode composed of a blend of active materials, for use with Single
     Particle type models.
     """
 
-    particle: Dict[str, Particle] = Field(alias="Particle")
+    particle: dict[str, Particle] = Field(alias="Particle")
 
 
 class UserDefined(BaseModel):
-    class Config:
-        extra = Extra.allow
+    model_config = ConfigDict(extra="allow")
 
-    def __init__(self, **data):
+    def __init__(self, **data: int) -> None:
```

Comment:
I am still looking at this spot. The type is marked as int, but it is treated like a dictionary below